sale_purchase_force_vendor/models/stock_move.py

```diff
@@ -0,0 +1,16 @@
+# Copyright 2022 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import models
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    def _prepare_procurement_values(self):
+        res = super()._prepare_procurement_values()
+        if self.sale_line_id.vendor_id:
+            res_order_line = self.sale_line_id._prepare_procurement_values(
+                group_id=self.group_id
+            )
+            res.update({"supplierinfo_id": res_order_line["supplierinfo_id"]})
+        return res
```

Comment:
The code is not correct, and please always add docstring for the WHYs. Is this case covered by tests?

```suggestion
        """Inject the preferred vendor in case of a MTO that first creates the OUT
        move.
        """
        res = super()._prepare_procurement_values()
        if self.sale_line_id.vendor_id:
            res["supplier_id"] = self.sale_line_id.vendor_id
        return res
```

Reply:
Ok sorry, the docstring is already added.
The test uses this, but `supplierinfo_id` field is required for the test to work properly and create the order as per https://github.com/odoo/odoo/blob/14.0/addons/purchase_stock/models/stock_rule.py#L57

Reply:
Please check, as this is not required in the previous code (but it was for v13).